node/metrics/src/tests.rs

```diff
@@ -92,14 +92,17 @@ async fn scrape_prometheus_metrics(metrics_uri: &str) -> HashMap<String, u64> {
 		.expect("Scraper failed to parse Prometheus metrics")
 		.samples
 		.into_iter()
-		.map(|sample| {
+		.map(|prometheus_parse::Sample { metric, value, labels, .. }| {
 			(
-				sample.metric.to_owned(),
-				match sample.value {
+				metric.to_owned(),
+				match value {
 					prometheus_parse::Value::Counter(value) => value as u64,
 					prometheus_parse::Value::Gauge(value) => value as u64,
 					prometheus_parse::Value::Untyped(value) => value as u64,
-					_ => unreachable!("unexpected metric type"),
+					value @ _ => unreachable!(
```

Comment:
I think it would be better to dummy handle the rest of variants and don't store anything in the hashmap for them.
- Histogram([Vec](https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html)<[HistogramCount](https://docs.rs/prometheus-parse/0.2.3/prometheus_parse/struct.HistogramCount.html)>),
 - Summary([Vec](https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html)<[SummaryCount](https://docs.rs/prometheus-parse/0.2.3/prometheus_parse/struct.SummaryCount.html)>)

This is currently used by runtime metrics, which AFAIK are not really used. However this code could be moved to a place where it can be used to assert node metrics as well.

In the future if we want to assert against histograms for example we could implement it, but for now I think it's better to not fail tests without good reason.

Reply:
@sandreim

this is falling in `origin/gav-xcm-v3` branch, master works

and the error was this:
```
test tests::runtime_can_publish_metrics ... FAILED

failures:

---- tests::runtime_can_publish_metrics stdout ----
thread 'tests::runtime_can_publish_metrics' panicked at 'internal error: entered unreachable code: unexpected metric type, value: Histogram([HistogramCount { less_than: 0.005, count: 0.0 }, HistogramCount { less_than: 0.01, count: 0.0 }, HistogramCount { less_than: 0.025, count: 0.0 }, HistogramCount { less_than: 0.05, count: 0.0 }, HistogramCount { less_than: 0.1, count: 0.0 }, HistogramCount { less_than: 0.25, count: 0.0 }, HistogramCount { less_than: 0.5, count: 0.0 }, HistogramCount { less_than: 1.0, count: 0.0 }, HistogramCount { less_than: 2.5, count: 0.0 }, HistogramCount { less_than: 5.0, count: 0.0 }, HistogramCount { less_than: 10.0, count: 0.0 }, HistogramCount { less_than: inf, count: 0.0 }]), sample_metric: "polkadot_parachain_subsystem_unbounded_tof", sample_labels: Labels({"subsystem_name": "bitfield-distribution-subsystem", "chain": "test-chain"})', node/metrics/src/tests.rs:104:34
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    tests::runtime_can_publish_metrics

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 20.16s
```

Reply:
@sandreim
I could remove that `unreachable` and just skip and dump,
just tell me :)

Reply:
I don't think we have to dump at all. But the panic is a mistake on my side when I wrote the code initially as I didn't take into account the other metric types the node could emit. 
Just skipping them is fine.